### PR TITLE
 Complex input factories

### PR DESCRIPTION
**Pull request by sebastian-peter**

### CODE REVIEW

**Comment by sebastian-peter** [comment]
!test

File: src/main/java/edu/ie3/models/GermanVoltageLevel.java [hunk 1]
```diff
@@ -46,4 +46,19 @@
   public String getName() {
     return name;
   }
+
+  public static VoltageLevel parseVoltageLvl(String value) {
+    switch (value) {
```
Comment (ckittl) [inline src/main/java/edu/ie3/models/GermanVoltageLevel.java]: What about `value.toLowerCase()`?

File: src/main/java/edu/ie3/io/factory/input/AssetInputEntityData.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Data used for the construction of {@link edu.ie3.models.input.AssetInput} entities. This data
+ * object includes additional information about the {@link OperatorInput}, which cannot be provided
+ * through the attribute map.
+ */
+public class AssetInputEntityData extends EntityData {
+  private final OperatorInput operatorInput;
+
+  /**
+   * Creates a new AssetInputEntityData object for a non-operable asset input
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/io/factory/input/AssetInputEntityData.java]: I got this from AssetInput.java, where constructors are called exactly this way. I don't have the knowledge to tell whether this wording makes sense or not.

File: src/main/java/edu/ie3/io/factory/input/AssetInputEntityFactory.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+   * Set[])}, integrity of operational data is checked here.
+   *
+   * @param data the entity containing at least the entity class as well a mapping of the provided
+   *     field name strings to its value (e.g. a headline of a csv -> column values)
+   * @param fieldSets a set containing all available constructor combinations as field names
+   * @return the index of the set in the fieldSets array that fits the provided entity data
+   */
+  @Override
+  protected int validateParameters(D data, Set<String>... fieldSets) {
+    if ((data.containsKey(OPERATES_FROM)
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/io/factory/input/AssetInputEntityFactory.java]: As discussed, I'm removing these checks now and from now on it is ok to only pass operation time _or_ operator input.

File: src/main/java/edu/ie3/models/GermanVoltageLevel.java (hunk 1 above)
Comment (sebastian-peter) [reply src/main/java/edu/ie3/models/GermanVoltageLevel.java]: Sounds good.

**Comment by johanneshiry** [comment]
!test

**Pull request merged**